source/common/router/config_impl.cc

```diff
@@ -602,6 +612,32 @@ RouteEntryImplBase::loadRuntimeData(const envoy::config::route::v3::RouteMatch&
   return runtime;
 }
 
+const std::string&
+RouteEntryImplBase::getPathRewrite(const Http::RequestHeaderMap& headers,
+                                   absl::optional<std::string>& container) const {
+  // Just use the prefix rewrite  if this isn't a redirect.
+  if (!isRedirect()) {
+    return prefix_rewrite_;
+  }
+
+  // Return the regex rewrite substitution for redirects, if set.
+  if (regex_rewrite_redirect_ != nullptr) {
+    std::string path(headers.getPathValue());
+
+    // Replace the entire path, but preserve the query parameters
+    auto just_path(Http::PathUtil::removeQueryAndFragment(path));
+
+    // Store the result in the output container, and return a reference to the underlying string.
+    container = std::move(path.replace(
```

Comment:
This is a little odd but it keeps the zero-copy const reference return value for the common (no regex rewrite) case.

Critical feedback welcome if there's another preferred way to do this.

Reply:
Yeah this is fine.